### PR TITLE
Add RazorRendering benchmarkapp

### PR DESCRIPTION
**Pull request by benaadams**
Sample page for performance testing as requested

Url Path: /Category/PageA

/cc @rynowak @davidfowl 

Call count tracing for 1,144 requests; items for PRs highlighted: 
* Diagnostics (https://github.com/aspnet/Mvc/pull/8349 not merged breaking)
* IList (https://github.com/aspnet/Mvc/pull/8353 merged)
* ViewBuffer (https://github.com/aspnet/Mvc/pull/8339 merged)
* Reduce viewContext (https://github.com/aspnet/Mvc/pull/8338 merged)

image

### CODE REVIEW

**Comment by benaadams** [comment]
Decompiled page sample
image

**Comment by benaadams** [comment]
One request (core 2.1)
image

**Comment by rynowak** [comment]
Awesome 👍

**Comment by rynowak** [comment]
We're currently locked down for the next few days while we prepare a preview 2 build. 

@dougbu  could you take a look at the project file since you did the other ones?

File: benchmarkapps/RazorRendering/RazorRendering.csproj [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  <ItemGroup Condition="'$(BenchmarksTargetFramework)' == ''">
+    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="$(MicrosoftAspNetCoreServerKestrelPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.CommandLine" Version="$(MicrosoftExtensionsConfigurationCommandLinePackageVersion)" />
+
+    <ProjectReference Include="..\..\src\Microsoft.AspNetCore.Mvc\Microsoft.AspNetCore.Mvc.csproj" />
+  </ItemGroup>
+
+  <!--
+    These references are used when running on the Benchmarks Server.
+    Use All meta-package and not App to include Microsoft.EntityFrameworkCore.Sqlite.
```
Comment (dougbu) [inline benchmarkapps/RazorRendering/RazorRendering.csproj]: This was needed in the Basic* apps. Doubt it's needed here. Try `Microsoft.AspNetCore.App`

File: benchmarkapps/RazorRendering/RazorRendering.csproj [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp2.2</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(DeveloperBuild)' != 'true' AND '$(OS)' == 'Windows_NT' ">$(TargetFrameworks)</TargetFrameworks>
+    <TargetFrameworks Condition="'$(BenchmarksTargetFramework)' != ''">$(BenchmarksTargetFramework)</TargetFrameworks>
+
+    <WarningsNotAsErrors>CS8002;$(WarningsNotAsErrors)</WarningsNotAsErrors>
```
Comment (dougbu) [inline benchmarkapps/RazorRendering/RazorRendering.csproj]: This shouldn't be needed in this project. Was required in the Basic* apps due to a dependency lacking a strong name.

**Comment by rynowak** [comment]
@benaadams FYI we're in lockdown mode to produce a build of preview2, so you can expect this to get merged sometime next week 😀

File: benchmarkapps/RazorRendering/RazorRendering.csproj (hunk 2 above)
Comment (benaadams) [reply benchmarkapps/RazorRendering/RazorRendering.csproj]: Changed

**Comment by pranavkm** [comment]
Thanks for the PR!

**Pull request merged**